PWGDQ/Core/VarManager.h

```diff
@@ -1029,6 +1041,144 @@ void VarManager::FillPairVertexing(C const& collision, T const& t1, T const& t2,
+
+    SMatrix5 t3pars(track.x(), track.y(), track.phi(), track.tgl(), track.signed1Pt());
+    std::vector<double> v3{track.cXX(), track.cXY(), track.cYY(), track.cPhiX(), track.cPhiY(),
+                           track.cPhiPhi(), track.cTglX(), track.cTglY(), track.cTglPhi(), track.cTglTgl(),
+                           track.c1PtX(), track.c1PtY(), track.c1PtPhi(), track.c1PtTgl(), track.c1Pt21Pt2()};
+    SMatrix55 t3covs(v3.begin(), v3.end());
+    o2::track::TrackParCovFwd pars3{track.z(), t3pars, t3covs, chi23};
+    procCode = VarManager::fgFitterThreeProngFwd.process(pars1, pars2, pars3);
+    procCodeJpsi = VarManager::fgFitterTwoProngFwd.process(pars1, pars2);
+  } else if constexpr ((pairType == kJpsiToEE) && trackHasCov) {
+    mtrack = fgkElectronMass;
+    std::array<float, 5> lepton1pars = {lepton1.y(), lepton1.z(), lepton1.snp(), lepton1.tgl(), lepton1.signed1Pt()};
+    std::array<float, 15> lepton1covs = {lepton1.cYY(), lepton1.cZY(), lepton1.cZZ(), lepton1.cSnpY(), lepton1.cSnpZ(),
+                                         lepton1.cSnpSnp(), lepton1.cTglY(), lepton1.cTglZ(), lepton1.cTglSnp(), lepton1.cTglTgl(),
+                                         lepton1.c1PtY(), lepton1.c1PtZ(), lepton1.c1PtSnp(), lepton1.c1PtTgl(), lepton1.c1Pt21Pt2()};
+    o2::track::TrackParCov pars1{lepton1.x(), lepton1.alpha(), lepton1pars, lepton1covs};
+    std::array<float, 5> lepton2pars = {lepton2.y(), lepton2.z(), lepton2.snp(), lepton2.tgl(), lepton2.signed1Pt()};
+    std::array<float, 15> lepton2covs = {lepton2.cYY(), lepton2.cZY(), lepton2.cZZ(), lepton2.cSnpY(), lepton2.cSnpZ(),
+                                         lepton2.cSnpSnp(), lepton2.cTglY(), lepton2.cTglZ(), lepton2.cTglSnp(), lepton2.cTglTgl(),
+                                         lepton2.c1PtY(), lepton2.c1PtZ(), lepton2.c1PtSnp(), lepton2.c1PtTgl(), lepton2.c1Pt21Pt2()};
+    o2::track::TrackParCov pars2{lepton2.x(), lepton2.alpha(), lepton2pars, lepton2covs};
+    std::array<float, 5> lepton3pars = {track.y(), track.z(), track.snp(), track.tgl(), track.signed1Pt()};
+    std::array<float, 15> lepton3covs = {track.cYY(), track.cZY(), track.cZZ(), track.cSnpY(), track.cSnpZ(),
+                                         track.cSnpSnp(), track.cTglY(), track.cTglZ(), track.cTglSnp(), track.cTglTgl(),
+                                         track.c1PtY(), track.c1PtZ(), track.c1PtSnp(), track.c1PtTgl(), track.c1Pt21Pt2()};
+    o2::track::TrackParCov pars3{track.x(), track.alpha(), lepton3pars, lepton3covs};
+    procCode = VarManager::fgFitterThreeProngBarrel.process(pars1, pars2, pars3);
+    procCodeJpsi = VarManager::fgFitterTwoProngBarrel.process(pars1, pars2);
+  } else {
+    return;
+  }
+
+  ROOT::Math::PtEtaPhiMVector v1(lepton1.pt(), lepton1.eta(), lepton1.phi(), mtrack);
+  ROOT::Math::PtEtaPhiMVector v2(lepton2.pt(), lepton2.eta(), lepton2.phi(), mtrack);
+  ROOT::Math::PtEtaPhiMVector v3(track.pt(), track.eta(), track.phi(), mtrack);
```

Comment:
This vector assumes that the track is an electron or a muon, but if we want to use this function for channels like B+ -> J/psi + K+ then we need to have a way of transmitting this to the function. This can be transmitted via the template argument you call now "pairType". This could be generalized to sometihng like "candidateType" and you can have some specific entries added to the "PairCandidateType" enumeration. For example, you could have there, in addition to what is there now, kBcToThreeMuons (your case for Bc to three muons), kBtoJpsiEEK (B+ -> jpsi + K+), etc. So based on these we can properly deduce the mass assumption for all three prongs using constexpr.

Reply:
Implemented with the new commit!